Upgrade Go from v1.24.6 -> 1.25.1

djjuhasz opened the pull request.

djjuhasz committed:
Upgrade Go from v1.24.6 -> 1.25.1

djjuhasz committed:
Upgrade tools

Upgrade Enduro tools with `bine upgrade` to fix compile errors with
the golangci-lint and workflowcheck CI checks due to the new Go version.

djjuhasz committed:
Synchronize tool and module versions

Upgrade go modules that have a corresponding command line tool so the
tool and module versions are sychronized.

- Upgrade ariga.io/atlas v0.32.0 -> v0.37.0
- Upgrade entgo.io/ent v0.14.4 -> v0.14.5
- Upgrade goa.design/goa/v3 v3.15.2 -> v3.22.2

jraddaoui reviewed (changes requested):
Thanks @djjuhasz! It looks like it needs something else for the OpenTelemetry upgrade:

```
enduro │ 2025-09-19T00:26:21.951Z	V(2)	enduro	enduro/main.go:113	Error creating tracer provider.	{"error": "TracerProvider: new resource: conflicting Schema URL: https://opentelemetry.io/schemas/1.34.0 and https://opentelemetry.io/schemas/1.26.0"}
enduro │ main.main
enduro │ 	github.com/artefactual-sdps/enduro/cmd/enduro/main.go:113
enduro │ runtime.main
enduro │ 	runtime/proc.go:285
```

djjuhasz committed:
Upgrade OpenTelemetry modules to 1.38.0

Fix an OpenTelemetry error due to conflicting semantic convention
versions by updating all otel modules for compatiblity with:

- go.opentelemetry.io/otel@v1.38.0
- go.opentelemetry.io/otel/semconv/v1.37.0

jraddaoui reviewed (approved):
Thanks!!
